core/block.ex

```diff
@@ -56,11 +62,25 @@ defmodule UltraDark.Blockchain.Block do
   """
   @spec mine(Block) :: Block
   def mine(block) do
-    %{index: index, previous_hash: previous_hash, timestamp: timestamp, nonce: nonce, merkle_root: merkle_root} = block
-
-
-    block = %{ block | hash: Utilities.sha3_base16([Integer.to_string(index), previous_hash, timestamp, Integer.to_string(nonce), merkle_root]) }
+    %{
+      index: index,
+      previous_hash: previous_hash,
+      timestamp: timestamp,
+      nonce: nonce,
+      merkle_root: merkle_root
+    } = block
 
+    block = %{
+      block
+      | hash:
```

Comment:
this makes it super hard to tell whats happening. lets revert it back to the original definition